x/txfees/keeper/feedecorator_test.go

```diff
@@ -157,23 +174,60 @@ func (suite *KeeperTestSuite) TestFeeDecorator() {
 			gasRequested: mempoolFeeOpts.HighGasTxThreshold,
 			isCheckTx:    true,
 			expectPass:   true,
+			baseDenomGas: false,
 		},
 	}
 
 	for _, tc := range tests {
-
 		suite.ctx = suite.ctx.WithIsCheckTx(tc.isCheckTx)
 		suite.ctx = suite.ctx.WithMinGasPrices(tc.minGasPrices)
 
-		tx := legacytx.NewStdTx([]sdk.Msg{}, legacytx.NewStdFee(
-			tc.gasRequested,
-			tc.txFee,
-		), []legacytx.StdSignature{}, "")
+		// TxBuilder components reset for every test case
+		txBuilder := suite.clientCtx.TxConfig.NewTxBuilder()
+		priv0, _, addr0 := testdata.KeyTestPubAddr()
+		acc1 := suite.app.AccountKeeper.NewAccountWithAddress(suite.ctx, addr0)
+		suite.app.AccountKeeper.SetAccount(suite.ctx, acc1)
+		msgs := []sdk.Msg{testdata.NewTestMsg(addr0)}
+		privs, accNums, accSeqs := []cryptotypes.PrivKey{priv0}, []uint64{0}, []uint64{0}
+		signerData := authsigning.SignerData{
+			ChainID: suite.ctx.ChainID(), 
+			AccountNumber: accNums[0], 
+			Sequence: accSeqs[0]}
+
+		gasLimit := tc.gasRequested
+		sigV2, _ := clienttx.SignWithPrivKey(
+			1, 
+			signerData,
+			txBuilder, 
+			privs[0], 
+			suite.clientCtx.TxConfig,
+			accSeqs[0])
+
+		simapp.FundAccount(suite.app.BankKeeper, suite.ctx, addr0, tc.txFee)
+		
+		txBuilder.SetMsgs(msgs[0])
+		txBuilder.SetSignatures(sigV2)
+		txBuilder.SetMemo("")
+		txBuilder.SetFeeAmount(tc.txFee)
+		txBuilder.SetGasLimit(gasLimit)
+
+		tx := txBuilder.GetTx()
```

Comment:
Thoughts on refactoring all of this to a helper method?

Reply:
https://github.com/osmosis-labs/osmosis/blob/main/app/apptesting/test_suite.go would be the perfect place, where we keep methods of useful helper methods

Reply:
Moved the transaction building component into a helper method! It might be worth eventually creating a "basic transaction" helper method that fills in blank parameters for everything and builds a test tx, but in this case the fee and gas limit changes in each test case so the only part that could be cleanly abstracted was the building of the tx.